docs/_templates/search.html

```diff
@@ -0,0 +1,109 @@
+{% extends "layout.html" %}
+{% set title = _('Search') %}
+
+{% block body %}
+  <noscript>
+  <div id="fallback" class="admonition warning">
+    <p class="last">
+      {% trans trimmed %}Please activate JavaScript to enable the search
+      functionality.{% endtrans %}
+    </p>
+  </div>
+  </noscript>
+
+  <div id="search-results">
+  </div>
+{% endblock %}
+
+{% block footer %}
+{{ super() }}
+<script>
+
+function runSearchPageSearch(event) {
+  const urlParams = new URLSearchParams(window.location.search)
+  const query = urlParams.get('q')
+
+  const searchResultsContainer = document.getElementById('search-results')
+  addHeadingForQuery(query, searchResultsContainer)
+
+  const docSearch = docSearchReady()
+  const index = docSearch.client.initIndex('wagtail')
+  index.search(
+    query,
+    {
+      hitsPerPage: 50,
+      facetFilters: [getVersionFacetFilter()]
+    }
+  )
+  .then(({ hits }) => addResultsList(hits, query, searchResultsContainer))
+  .catch((error) => console.log(error))
+}
+
+function addHeadingForQuery(query, parentElement) {
+  const searchHeading = document.createElement('h2')
+  searchHeading.textContent = `Search Results for "${query}"`
+  parentElement.appendChild(searchHeading)
+}
+
+function addResultsList(hits, query, parentElement) {
+  const searchResultsList = document.createElement('ul')
+  searchResultsList.className = "search"
+  for (hit of hits) {
+    const hitElement = createHitElement(hit, query)
+    searchResultsList.appendChild(hitElement)
+  }
+  parentElement.appendChild(searchResultsList)
+}
+
+function createHitElement(hitData, query) {
+  const pageURL = extractPageLink(hitData.url) + '?highlight=' + query
+  const anchor = hitData.anchor
+  const anchorURL = pageURL + '#' + anchor
+  const result = hitData._highlightResult
+
+  const hitListElement = document.createElement('li')
+
+  let firstHierarchyLevel = undefined
+  let lastHierarchyLevel = undefined
+  for (levelLabel in result.hierarchy) {
+    if (firstHierarchyLevel === undefined ) {
+      firstHierarchyLevel = result.hierarchy[levelLabel]
+    }
+    lastHierarchyLevel = result.hierarchy[levelLabel]
+  }
```

Comment:
I’m not entirely sure why this is needed (would warrant a comment) but feels overkill to get the first and last item.

```suggestion
  const hierarchies = Object.values(result.hierarchy);
  const firstHierarchyLevel = hierarchies[0];
  const lastHierarchyLevel = hierarchies[hierarchies.length - 1];
```

Reply:
Oh yes. That could have used some refactoring. I thought at first that the properties would exist but empty. 

Your way is obviously way more elegant!